### PR TITLE
feat: add phpMyAdmin service configuration

### PR DESCRIPTION
**Pull request by monteshot**
- Add docker-compose.phpmyadmin.yml for phpMyAdmin service
- Update svc.cmd to conditionally include phpMyAdmin configuration
- Ensure phpMyAdmin config file is generated if not present
- Add regeneratePMAConfig function to core.sh
- Call regeneratePMAConfig on env changes in env.cmd
- Include phpMyAdmin as a global service in core.sh

### CODE REVIEW

**Comment by monteshot** [comment]
image

**Comment by bap14** [comment]
@monteshot Thanks for the contribution. I would like to suggest that you add a feature flag to this, similar to that of Portainer and DNSMasq, so that those that use external tools like DBeaver or TablePlus can opt-out of running PMA.

**Comment by poespas** [comment]
I notice that it creates the file in the global warden home directory, does this mean that when you have more than 1 running environment, it will override the first configuration?

We use PHPMyAdmin in our Warden environments, and use the `PMA_HOST` `PMA_USER` `PMA_PASSWORD` environment variables for this so we don't need to regenerate a configuration file and keep it within docker.

**Comment by monteshot** [comment]
@bap14 Hello. Please review the updated PR.

@poespas Hi. If you use PMA by extending the warden-env.yml file in the project directory, you will have two separate PMA apps. Regarding the file in the Warden home directory, this file only needs to give information about db containers(as mentioned in PMA [documentation](https://docs.phpmyadmin.net/en/latest/setup.html#customizing-configuration-file-using-docker-compose)).

**Comment by monteshot** [comment]
Updated the PR with the latest changes from the main branch. 
Additionally, I added one more DB to the hidden database list.

@bap14 @navarr, please review and merge the PR when you are ready.
:handshake:

File: commands/svc.cmd [hunk 1]
```diff
@@ -64,6 +72,12 @@ if [[ "${WARDEN_PARAMS[0]}" == "up" ]]; then
     mkdir -p "${WARDEN_HOME_DIR}/etc/traefik"
     cp "${WARDEN_DIR}/config/traefik/traefik.yml" "${WARDEN_HOME_DIR}/etc/traefik/traefik.yml"
 
+    if [[ "${WARDEN_PHPMYADMIN_ENABLE}" == 1 ]]; then
+        if [[ ! -f "${WARDEN_HOME_DIR}/etc/phpmyadmin/config.user.inc.php" ]]; then
+            mkdir -p "${WARDEN_HOME_DIR}/etc/phpmyadmin"
+        fi
+    fi
```
[the remark this answers is not in the document]
Comment (monteshot) [reply commands/svc.cmd:79]: 1. The main reason to leave PMA file generation here is faulting to build by docker-compose 
2. ++

**Comment by monteshot** [comment]
@ihor-sviziev Can you review PR again?

**Review by ihor-sviziev — APPROVED** [review]

**Comment by ihor-sviziev** [comment]
@navarr @bap14 are there any reasons not to merge this?

**Comment by monteshot** [comment]
Updated the PR with the latest changes from the main branch.

@bap14 @navarr

**Comment by hardyjohnson** [comment]
Can we re-open this contribution for discussion as implemented? It is possible I just don't understand the problem this is supposed to solve, mainly why is this seemingly forced into the global svc level vs implementing at the environments level?

1. I believe the original intent of global svc level services were for common and shared internet protocol services (dns, routing, mail, sshd, etc.) not application level services which are env and stack specific.

1. This is enabled by default, why not opt-in vs opt-out? (https://github.com/monteshot/warden/blob/fe5fedbb14180008e09e42e26b2bad555b1a3483/commands/install.cmd#L97)?

1. As implemented this introduces a configuration file as a parsed string into the core util, this might be better served as a file mapped configuration file at the environments level leveraging environment variables as per other services.

1. There is currently a bug where the regenerate PMA config command is run for every invocation of the warden svc command. e.g. `warden svc ps`, see #852. 

@bap14 @navarr @ihor-sviziev @monteshot

**Comment by navarr** [comment]
@hardyjohnson Please open this up as a discussion in the forums under Ideas so it's easier not to lose. I'll reply there. I'm glad you brought this up, and I can see several ways to improve this

**Pull request merged**